nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
It could be adapted if that happens, but it has remained stable for a loong time. It's more likely to go away due to them redoing the weak handles again. They're already getting rid of them in favor of phantom persistents.

On May 18, 2015 10:46:34 PM EEST, Ben Noordhuis <notifications@github.com> wrote:
>>    typedef void (*Callback)(
>>        NanWeakCallbackData & data  // NOLINT(runtime/references)
>>        );
>> -  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
>> -    : parameter(param), callback(cb) {
>> -    NanAssignPersistent(persistent, handle);
>> -    Revive();
>> +  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback
>cb)
>> +    : persistent(), parameter(param), callback(cb) {
>> +    std::memcpy(&persistent, handle, sizeof(persistent));
>
>The pertinent word that is missing in your last sentence is
>'currently'. :-)  It may work not but even a trivial V8 upgrade could
>break it.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/348/files#r30540758